lit.ts

```
@@ -478,6 +491,173 @@ export async function getPubkeyForAuthMethod({
+	const dataToString = Buffer.from(data, 'base64').toString('binary');
+
+	return JSON.parse(dataToString);
+}
+
+function normalizeCapacity(capacity: any) {
+	const [requestsPerMillisecond, expiresAt] = capacity[0];
+
+	return {
+		requestsPerMillisecond: parseInt(requestsPerMillisecond.toString()),
+		expiresAt: {
+			timestamp: parseInt(expiresAt.toString()),
+		},
+	};
+}
+
+async function queryCapacityCredit(contract: ethers.Contract, tokenId: number) {
+	const [URI, capacity, isExpired] = await Promise.all([
+		contract.functions.tokenURI(tokenId).then(normalizeTokenURI),
+		contract.functions.capacity(tokenId).then(normalizeCapacity),
+		contract.functions.isExpired(tokenId),
+	]);
+
+	return {
+		tokenId,
+		URI,
+		capacity,
+		isExpired,
+	} as CapacityToken;
+}
+
+export async function queryCapacityCredits(signer: ethers.Wallet) {
+	if (config.network === "serrano" || config.network == "cayenne") {
+		throw new Error("Capacity credits are not available on Serrano");
+	}
+
+	const contract = await getContractFromWorker(config.network, 'RateLimitNFT');
+	const count = parseInt(await contract.functions.balanceOf(signer.address));
+
+	return await Promise.all([...new Array(count)].map(async (_, i) => {
+		return queryCapacityCredit(contract, i);
+	})) as CapacityToken[];
+}
+
+export async function addPaymentDelegationPayee({
+	wallet, payeeAddresses
+}: {
+	wallet: ethers.Wallet;
+	payeeAddresses: string[];
+}) {
+	if (config.network === "serrano" || config.network == "cayenne") {
+		throw new Error(`Payment delegation is not available on ${config.network}`);
+	}
+
+	// TODO: It would be good to just implement the logic for this locally, and avoid
+	//		 having to pull in the LitNodeClientNodeJs package.
+	const client = new LitNodeClientNodeJs({
+		alertWhenUnauthorized: false,
+		litNetwork: config.network,
+	});
+
+	let capacityTokens: CapacityToken[] = [];
+
+	try {
+		capacityTokens = (await queryCapacityCredits(wallet));
```

Why do we need scoping around `queryCapacityCredits`

---

It was to separate no credits vs unable to fetch credits, but it does somewhat hide the error as it will fall through and then try to mint a new one. I'll remove it and just let it throw the error up to the response.